File: Data Structures & Algorithms/k-closest-points-to-origin/submission-5.cpp
class Solution {
public:
    float distance(int a, int b){
        float d = sqrt((float)(a*a+b*b));
        return d;
    }
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        vector<vector<int>> ans;
        priority_queue<pair<float,int>>pq;
        for(int i=0;i<points.size();i++){
            float d = distance(points[i][0],points[i][1]);
            cout<<d<<endl;

            if(pq.size()<k){
                pq.push(make_pair(d,i));
            }
            else{
                cout<<d<<" "<<pq.top().first<<endl;
                if(pq.top().first>d){
                    cout<<"hellp"<<endl;
                    pq.pop();
                    pq.push(make_pair(d,i));
                }
            }
        }

        while(!pq.empty()){
            ans.push_back(points[pq.top().second]);
            pq.pop();
        }
        return ans;


    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-5.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st = []
        ans = [0]*len(temperatures)
        i=len(temperatures)-1
        while i>=0:
            while len(st)!=0 and temperatures[i]>=temperatures[st[-1]]:
                st.pop()
            if len(st)==0:
                ans[i] = 0
            else:
                ans[i] = (st[-1]-i)
            
            st.append(i)
            i-=1

        return ans

            